bioframe/core.py

```diff
@@ -0,0 +1,684 @@
+        if raise_errors:
+            raise TypeError("Invalid columns")
+        return False
+
+    if not _verify_column_dtypes(df, cols=[ck1, sk1, ek1], return_as_bool=True):
+        if raise_errors:
+            raise TypeError("Invalid column dtypes")
+        return False
+
+    if ((df[ek1] - df[sk1]) < 0).any():
+        if raise_errors:
+            raise ValueError(
+                "Invalid genomic interval dataframe: starts exceed ends for "
+                + str(np.sum(((df[ek1] - df[sk1]) < 0)))
+                + " intervals"
+            )
+        return False
+
+    return True
+
+
+def is_overlapping(df, cols=None):
+    """
+    tests if any genomic intervals in a bioframe `df` overlap
+
+    Returns
+    -------
+    is_overlapping:bool
+
+    """
+
+    ck1, sk1, ek1 = _get_default_colnames() if cols is None else cols
+
+    df_merged = ops.merge(df, cols=cols)
+
+    total_interval_len = np.sum((df[ek1] - df[sk1]).values)
+    total_interval_len_merged = np.sum((df_merged[ek1] - df_merged[sk1]).values)
+
+    if total_interval_len > total_interval_len_merged:
+        return True
+    else:
+        return False
+
+
+def is_viewframe(region_df, raise_errors=False, view_name_col="name", cols=None):
+    """
+    Checks that region_df is a valid view, namely that it:
+    - satisfies requirements for a bedframe, including columns for ('chrom', 'start', 'end')
+    - has an additional column providing a unique name, view_name_col, with default 'name'
+    - does not contain null values
+    - that entries in the view_name_col are unique.
+    - intervals are non-overlapping
+
+    raise_errors:bool
+        If true, raises errors instead of returning a boolean for invalid properties.
+        Default false.
+
+    Returns
+    -------
+    is_viewframe:bool
+
+    """
+
+    ck1, sk1, ek1 = _get_default_colnames() if cols is None else cols
+
+    if not _verify_columns(
+        region_df, [ck1, sk1, ek1, view_name_col], return_as_bool=True
+    ):
+        if raise_errors:
+            raise TypeError("Invalid view: column names cannot be verified")
+        return False
+
+    if not is_bedframe(region_df, cols=cols):
+        if raise_errors:
+            raise ValueError("Invalid view: bedframe properties cannot be verified")
+        return False
+
+    if pd.isna(region_df).values.any():
+        if raise_errors:
+            raise ValueError("Invalid view: cannot contain NAs")
+        return False
+
+    if len(set(region_df[view_name_col])) < len(region_df[view_name_col].values):
+        if raise_errors:
+            raise ValueError(
+                "Invalid view: entries in region_df[view_name_col] must be unique"
+            )
+        return False
+
+    if is_overlapping(region_df, cols=cols):
+        if raise_errors:
+            raise ValueError("Invalid view: entries must be non-overlapping")
+        return False
+
+    return True
+
+
+def make_viewframe(
```

Comment:
this duplicates functionality of the [`bioframe.parse_regions`](https://github.com/open2c/bioframe/blob/0508270bac18ef15bdea9535d9126c958d6c9952/bioframe/region.py#L160) by @mimakaev  - we should merge these 2 functions, by bringing some of the functionality from `parse_regions` to `make_viewframe`:
 - force UCSC renaming ...
 - fill_missing ?
 - check_bounds ?

Also from the discussion - drop support for `{str:(int,int)}` format and `p/q` stripping ...

Reply:
to specify - functionality from `parse_regions` that we actually use in cooltools:
 - make the function check bounds of regions in a view by providing a dict-like for chromsizes
 - add `name` to regions even if they do not have one, by default force UCSC formatting, or allow a formatting function e.g. `lambda chrom, start, end: f"{chrom}:{start}-{end}"`

Not sure about `fill_missing` - never used it

Reply:
add legacy format for `view`(`regions`) - `[(chom, start, end)]` - and/or  - `[(chom, start, end, name)]` - potentially copy/paste from `bioframe.parse_regions`

Reply:
actually, seems like old check_bounds only check starts < ends... added an option to pass a dictlike chromsizes anyways